pbj-core/pbj-runtime/src/main/java/com/hedera/pbj/runtime/io/buffer/BufferedData.java

```diff
@@ -1017,4 +1040,2139 @@ public void writeVarLong(long value, final boolean zigZag) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            ReadOnlyByteBuffer that = (ReadOnlyByteBuffer) o;
+            if (this.capacity() != that.capacity()) return false;
+            if (this.limit() != that.limit()) return false;
+            for (int i = start; i < this.limit(); i++) {
+                if (buffer[i] != that.buffer[i]) return false;
+            }
+            return true;
+        }
+
+        /**
+         * Get hash based on contents of this buffer
+         *
+         * @return hash code
+         */
+        @Override
+        public int hashCode() {
+            return buffer.hashCode();
+        }
+
+        // ================================================================================================================
+        // SequentialData Methods
+
+        /**
+         * Get the capacity in bytes that can be stored in this buffer
+         *
+         * @return capacity in bytes
+         */
+        @Override
+        public long capacity() {
+            return limit;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public long position() {
+            return pos;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public long limit() {
+            return limit;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void limit(final long limit) {
+            throw new UncheckedIOException(new IOException("limit - Read-only buffer."));
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public boolean hasRemaining() {
+            return pos < limit;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public long remaining() {
+            return limit - pos;
+        }
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public long skip(long count) {
+            count = Math.min(count, remaining());
+            if (count <= 0) {
+                return 0;
+            }
+
+            pos = pos + (int) count;
+            return count;
+        }
+
+        // ================================================================================================================
+        // BufferedSequentialData Methods
+
+        /**
+         * {@inheritDoc}
+         */
+        @Override
+        public void position(final long position) {
+            if (position > limit)
+                throw new BufferOverflowException();
+
+            pos = Math.toIntExact(position);
+        }
+
+        /**
+         * Set the limit to current position and position to origin. This is useful when you have just finished writing
+         * into a buffer and want to flip it ready to read back from.
+         */
+        @Override
+        public void flip() {
+            pos = start;
```

Comment:
What about the limit? It should be set to `pos`, shouldn't it?

Reply:
Got catch. I thought I fixed that earlier.